includes/classes/shopping_cart.php

```diff
@@ -745,11 +745,11 @@ function calculate() {
           $sale_maker_discount = '';
 
           // bottom total
-          if ($attribute_price->fields['product_attribute_is_free'] == '1' and zen_get_products_price_is_free((int)$prid)) {
+          if (!$attribute_price->EOF && $attribute_price->fields['product_attribute_is_free'] == '1' and zen_get_products_price_is_free((int)$prid)) {
```

Comment:
Would it be safe to move these `! EOF` checks up to line 739 as a single `if ($attribute_price->EOF) continue;` ?

Reply:
So couple of things.
First, I had not stepped back to truly evaluate the possibility of "merging" these changes into a single `continue` operation; however, I had considered if that evaluation should be done and the excuse I identified at the time was, what if there were notifiers incorporated in this loop that truly expected to be executed (because previous versions supported stepping through each line of the loop).  There currently are no such notifiers in this area and the apparent purpose of this code section supports that simplification, but that was a reason that I "released" these changes for review.  It appears though that in absence of that possibility/concern that the current code supports the recommended simplification of continuing through the loop if there is no price data associated. (All occurrences of adding to the total resolve to adding 0 to an existing value).

I do not really have a preference either way though I also am not sure about modifications made by others to this code/area of code...